ksqldb-common/src/main/java/io/confluent/ksql/schema/ksql/SystemColumns.java

```diff
@@ -134,28 +143,58 @@ public static boolean isSystemColumn(final ColumnName columnName, final KsqlConf
     return isSystemColumn(columnName, getPseudoColumnVersionFromConfig(ksqlConfig));
   }
 
-  @SuppressFBWarnings(
-      value = "MS_EXPOSE_REP",
-      justification = "SYSTEM_COLUMN_NAMES is ImmutableSet"
-  )
-  private static Set<ColumnName> systemColumnNames(final int pseudoColumnVersion) {
-    return SYSTEM_COLUMN_NAMES_BY_VERSION.get(pseudoColumnVersion);
+  public static Set<ColumnName> systemColumnNames(final int pseudoColumnVersion) {
+    return Stream.concat(
+        WINDOW_BOUNDS_COLUMN_NAMES.stream(),
+        pseudoColumns
+            .stream()
+            .filter(col -> col.version <= pseudoColumnVersion)
+            .map(col -> col.name)
+    )
+        .collect(Collectors.toSet());
   }
 
   public static boolean mustBeMaterializedForTableJoins(final ColumnName columnName) {
-    return MUST_BE_MATERIALIZED_FOR_TABLE_JOINS.contains(columnName);
+    return pseudoColumns
```

Comment:
I'm confused about the contract of this method (`mustBeMaterializedForTableJoins`). The previous implementation did not throw if called on a non-pseudocolumn, but this new implementation does. Was this change in behavior from an earlier version of the PR intentional? Let's add a javadoc to clarify.

Reply:
Yeah, I changed the contract as I think it aligns more with the general philosophy of the future methods which will be similar: the extra properties I'm adding to pseudocolumns in the next 2 PRs were best served by throwing exceptions if a non-system column name was provided.